CoreRPC/Transport/NamedPipe/NamedPipeHost.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.IO.Pipes;
+using System.Threading.Tasks;
+
+namespace CoreRPC.Transport.NamedPipe
+{
+    public sealed class NamedPipeHost : IDisposable
+    {
+        private readonly IRequestHandler _engine;
+        private NamedPipeServerStream _pipe;
+        private bool _isDisposed;
+
+        public NamedPipeHost(IRequestHandler engine) => _engine = engine;
+
+        public void StartListening(string pipeName) => Task.Run(async () =>
+        {
+            while (!_isDisposed)
+            {
+                using (_pipe = new NamedPipeServerStream(pipeName, PipeDirection.InOut))
+                {
+                    await _pipe.WaitForConnectionAsync();
+
+                    var requestLengthBytes = await _pipe.ReadExactlyAsync(4);
```

Comment:
Spawn a new `Task` and do reading there, right now processing of new connections won't happen until the request is completely handled. Named pipe API is imperfect and this is the recommended way of dealing with multiple clients, see https://docs.microsoft.com/en-us/windows/desktop/ipc/multithreaded-pipe-server

Reply:
Should we read and write in a separate `Task.Run` or is reading enough in this particular case?

Reply:
Feels like resolved this, but needs a review.